redbot/core/commands/converter.py

```diff
@@ -116,3 +205,81 @@ class ValidatedConverter(DictConverter, metaclass=PartialMeta):
         pass
 
     return ValidatedConverter
+
+
+class TimedeltaConverter(dpy_commands.Converter):
+    """
+    This is a converter for timedeltas.
+    The units should be in order from largest to smallest.
+    This works with or without whitespace.
+
+    See `parse_timedelta` for more information about how this functions.
+
+    Attributes
+    ----------
+    maximum : Optional[timedelta]
+        If provided, any parsed value higher than this will raise an exception
+    minimum : Optional[timedelta]
+        If provided, any parsed value lower than this will raise an exception
+    allowed_units : Optional[List[str]]
+        If provided, you can constrain a user to expressing the amount of time
+        in specific units. The units you can chose to provide are the same as the
+        parser understands. `weeks` `days` `hours` `minutes` `seconds`
+    """
+
+    def __init__(self, *, minimum=None, maximum=None, allowed_units=None):
```

Comment:
Hmm, what about adding something along the lines of a `default_unit`, that would be used when the argument parsed would be an integer only?

This would be really helpful for other cogs where passing just a number would default to the unit specified, while passing numbers with units would be converted as it currently is now.

The argument could easily be checked with `argument.isdigit()` to determine if it should be parsed with `default_unit` or not.

Finally, setting `default_unit` to `None` (probably by default) would disable this additional functionality altogether.

Reply:
Is there any benefit to this beyond not using a union converter? I don't think guessing or assuming units in a converter designed to accept multiple is a good idea.

Reply:
I somewhat agree, except in the case of being guaranteed a specific type rather either using an int converter or needing to write a custom converter on top of this for that case.

I'm not completely made up on this either way, and think that while it's worth discussion, it's not worth delaying the addition since it can be added non-breakingly later, and in the mean time there are other options for this specific case.